### PR TITLE
Publish Java package from Github actions

### PR DESCRIPTION
**Pull request by pakrym-stripe**
Adds steps to release Java package and JavaDoc on tag pushes.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -93,3 +94,62 @@ jobs:
...
+        env:
+          GPG_SIGNING_PRIVKEY: ${{ secrets.GPG_SIGNING_PRIVKEY }}
+          GPG_SIGNING_PASSPHRASE: ${{ secrets.GPG_SIGNING_PASSPHRASE }}
+
+      - name: Publish package
+        run: |
+          ./gradlew publish closeAndRelease -Dorg.gradle.project.NEXUS_USERNAME=$NEXUS_USERNAME -Dorg.gradle.project.NEXUS_PASSWORD=$NEXUS_PASSWORD -Dorg.gradle.project.signing.gnupg.keyName=$GPG_SIGNING_KEYID -Dorg.gradle.project.signing.gnupg.passphrase=$GPG_SIGNING_PASSPHRASE
+        env:
+          TWINE_USERNAME: ${{ secrets.NEXUS_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
```
Comment (dcr-stripe) [inline .github/workflows/ci.yml:129]: TWINE -> NEXUS

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -93,3 +94,62 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@master
+      - name: Setup Java
+        id: setup-jre
+        uses: actions/setup-java@v1
+        with:
+          java-version: "18" # always use 18 for building
+          architecture: x64
+      - name: Set Test Java Runtime Environment variable
+        run: echo "JAVA_TEST_HOME=${{ steps.setup-jre.outputs.path }}" >> $GITHUB_ENV
```
Comment (dcr-stripe) [inline .github/workflows/ci.yml:114]: I wish we could share this across workflows :')

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -93,3 +94,62 @@ jobs:
         env:
           CI_NAME: github-actions
           COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+
+  publish:
+    if: >-
+        ((github.event_name == 'workflow_dispatch') || (github.event_name == 'push')) &&
+        (github.ref == 'refs/heads/pakrym/release-pipeline' || startsWith(github.ref, 'refs/tags/v')) &&
```
Comment (dcr-stripe) [inline .github/workflows/ci.yml:101]: Are you still testing this on your branch? If so let's skip the tag push for now and only do this on pushes to your branch?

**Review by dcr-stripe — COMMENTED** [review]
Small changes and then looks good to me! Once we're migrated to this we'll no longer have an asset tied to the GH release right?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (pakrym-stripe) [reply .github/workflows/ci.yml:114]: Same.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (pakrym-stripe) [reply .github/workflows/ci.yml:99]: Removed the tag trigger.

**Comment by pakrym-stripe** [comment]
PTAL @dcr-stripe ?

**Review by dcr-stripe — APPROVED** [review]
LGTM!

**Pull request merged**